spirv_msl.cpp

```diff
@@ -3602,7 +3606,7 @@ uint32_t CompilerMSL::add_interface_block(StorageClass storage, bool patch)
 		// the struct containing them is the correct size and layout.
 		for (auto &input : inputs_by_location)
 		{
-			if (location_inputs_in_use.count(input.first) != 0)
+			if (location_inputs_in_use.count(input.first.location) != 0)
```

Comment:
Here we lose component information. is_msl_shader_input_used() is missing a component member as well.

Reply:
To stay compatible I think we'd need two entry points. One where we return true if the Location is used at all (any component is fine) and one where we specify the matching Component as well, i.e. a narrower match.

Reply:
Hmmm...this was actually deliberate. 

I took the perspective that location is the lowest granularity of independence, analogous to a pixel, and if any component of a location is used, the entire location is considered used. Consequently, I didn't see a need to track usage below the level of a location.

But my assumptions might be wrong. Is there a usecase to track component usage?

Reply:
For tessellation inputs where we need to synthesize fake inputs, we'd need to know at a per-variable granularity, and that would require per-component granularity usage.

Reply:
I'm not 100% sure, but if you don't see a problem with this on your end in MoltenVK, I guess I can leave it for now. Can always be fixed later.